src/main/java/com/google/graphgeckos/dashboard/storage/GCDataRepository.java

```diff
@@ -0,0 +1,139 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.storage;
+
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+import com.google.cloud.datastore.TimestampValue;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.Date;
+import java.util.List;
+import org.springframework.boot.SpringApplication;
+import org.springframework.cloud.gcp.data.datastore.core.DatastoreTemplate;
+import org.springframework.stereotype.Repository;
+
+/**
+ * Implementation of the DataRepository, with the purpose of providing I/O to the Google Cloud
+ * Datastore. Each individual database entry is modeled by the {@link #BuildInfo BuildInfo} class.
+ * The relevant fields for the database, which are described in that class are:
+ *    - Kind: "revision"
+ *    - Key: commit hash
+ * The database indexes after all the BuildInfo fields, besides the {@code builders} list.
+ * It is unfeasable to query after the list of builders, since it would not be possible to
+ * replicate such a list without querying the database for it first, and the complexity
+ * of the indexing and querying would be great, without any apparent value.
+ *
+ * Useful links:
+ * - https://googleapis.dev/java/google-cloud-datastore/latest/index.html
+ * - https://googleapis.dev/java/spring-cloud-gcp/1.2.2.RELEASE/index.html
+ * - https://cloud.google.com/datastore/docs/concepts/
+ * - https://spring.io/projects/spring-cloud-gcp#overview
+ */
+@Repository
+public class GCDataRepository implements DataRepository {
```

Comment:
```suggestion
public class DatastoreRepository implements DataRepository {
```
The "GC" abbreviation in GCDataRepository is a bit unclear (naively I read it as "garbage-collected repository at first). I'd suggest using the name of the product (Google Datastore, or just Datastore) as a basis for the name of this class.